snc.sh

```diff
@@ -35,6 +36,64 @@ else
     fi
 fi
 
+function preflight_failure() {
+        local msg=$1
+        echo "$msg"
+        if [ -z "${SNC_NON_FATAL_PREFLIGHT_CHECKS-}" ]; then
+                exit 1
+        fi
+}
+
+function run_preflight_checks() {
+        echo "Checking libvirt and DNS configuration"
+
+        LIBVIRT_URI=qemu+tcp://localhost/system
+
+        # check if libvirtd is listening on a TCP socket
+        if ! virsh -c ${LIBVIRT_URI} uri >/dev/null; then
+                preflight_failure  "libvirtd is not listening for plain-text TCP connections, see https://github.com/openshift/installer/tree/master/docs/dev/libvirt#configure-libvirt-to-accept-tcp-connections"
+        fi
+
+        # check if libvirtd has access to an x86_64 hypervisor
+        if ! virsh -c ${LIBVIRT_URI} capabilities | ${XMLSTARLET} sel -t -c '/capabilities/guest/arch[@name="x86_64"]' >/dev/null; then
+                preflight_failure "Your platform does not provide a x86_64 hypervisor, SNC has only been tested on x86_64"
+                return
+        fi
+
+        # check if libvirtd x86_64 hypervisor can use kvm
+        if ! virsh -c ${LIBVIRT_URI} capabilities | ${XMLSTARLET} sel -t -c '/capabilities/guest/arch[@name="x86_64"]/domain[@type="kvm"]' >/dev/null; then
+                preflight_failure "Your platform does not provide a hardware-accelerated hypervisor, it's strongly recommended to enable it before running SNC. Check virt-host-validate for more detailed diagnostics"
+                return
+        fi
+
+        # check that api.crc.testing either can't be resolved, or resolves to 192.168.126.1[01]
+        local ping_status
+        ping_status="$(ping -c1 api.crc.testing | head -1 || true >/dev/null)"
+        if echo ${ping_status} | grep "PING api.crc.testing (" && ! echo ${ping_status} | grep "192.168.126.1[01])"; then
+                preflight_failure "DNS setup seems wrong, api.crc.testing resolved to an IP which is neither 192.168.126.10 nor 192.168.126.11, please check your NetworkManager configuration and /etc/hosts content"
+                return
+        fi
+
+        # check if firewalld is configured to allow traffic from 192.168.126.0/24 to 192.168.122.1
+        # this check is very basic and expects the configuration to match
+        # https://github.com/openshift/installer/tree/master/docs/dev/libvirt#firewalld
+        local zone
+        if firewall-cmd -h >/dev/null; then
+                # With older libvirt, the 'libvirt' zone will not exist
+                if firewall-cmd --get-zones |grep '\<libvirt\>'; then
+                        zone=libvirt
+                else
+                        zone=dmz
+                fi
```

Comment:
what if a user have created new zone manually and configure the libvirt traffic to that zone or we just want a check here which is listed to the https://github.com/openshift/installer/tree/master/docs/dev/libvirt#firewalld doc?

Reply:
I'm aware of this limitation which is mentioned in the comment before this test. I'm not sure many people are going to do a different configuration :) If you have suggestions for alternate ways of testing this, I can try a different test, I was short on ideas on how to achieve this one.